game/gamestate.go

```diff
@@ -41,10 +55,108 @@ func (gs *GameState) Copy() *GameState {
 	copy(ownershipCopy, gs.Ownership)
 
 	return &GameState{
-		Map:         gs.Map,
-		TroopCounts: troopCountsCopy,
-		Ownership:   ownershipCopy,
+		Map:           gs.Map,
+		TroopCounts:   troopCountsCopy,
+		Ownership:     ownershipCopy,
+		Rules:         gs.Rules,
+		CurrentPlayer: gs.CurrentPlayer,
+		LastMove:      gs.LastMove,
+	}
+}
+
+// LegalMoves returns all legal moves for the current player.
+func (gs *GameState) LegalMoves() []Move {
+	switch gs.Phase {
+	case ReinforcementPhase:
+		return gs.reinforcementMoves()
+	case AttackPhase:
+		return gs.attackMoves()
+	case ManeuverPhase:
+		return gs.maneuverMoves()
+	default:
+		return nil
+	}
+}
+
+// reinforcementMoves generates all possible reinforcement moves for the current player.
+func (gs *GameState) reinforcementMoves() []Move {
+	var moves []Move
+	// Calculate the number of troops to place
+	numTerritories := 0
+	for _, owner := range gs.Ownership {
+		if owner == gs.CurrentPlayer {
+			numTerritories++
+		}
+	}
+	troopsToPlace := max(3, numTerritories/3)

```

Comment:
player also gets troops from 1) auto-exchange cards 2) continent occupation
please refer to Jean's code for correctness and consistency.